repro.in

```diff
@@ -158,13 +158,19 @@ lock_close() {
 # 2: Command to execute
 function exec_nspawn(){
     local container=$1
+    if ((EPHEMERAL)); then
+      slock 8 "$BUILDDIRECTORY/$build.lock"
+    fi
     systemd-nspawn -q \
       --as-pid2 \
       --register=no \
       ${EPHEMERAL:+--ephemeral} \
       --pipe \
       -E "PATH=/usr/local/sbin:/usr/local/bin:/usr/bin" \
       -D "$BUILDDIRECTORY/$container" "${@:2}"
+    if ((EPHEMERAL)); then
+      lock_close 8 "$BUILDDIRECTORY/$build.lock"
```

Comment:
lock_close takes only one arg - the fd.

Although if the symmetric lock/unlock signature helps, let's keep it as-is. Might even extend lock_close to check fd against the filename, in the future.